### PR TITLE
geoclue2: 2.4.3 -> 2.4.7 and add wrapGAppsHook

### PR DESCRIPTION
**Pull request by ArdaXi**
###### Motivation for this change
geoclue2 on my machine failed to contact Mozilla because it was unable to find glib_networking. On GNOME systems this will be set by environment variable, but without GNOME it isn't. Adding `glib_networking` via the `wrapGAppsHook` solved this issue.

Additionally, when geoclue2 found an NMEA source on the local network via Avahi, it would crash on `No GSettings schemas are installed on the system` without that hook.

This PR solves both problems, as well as upgrading the package to the latest upstream.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/geoclue/2.0.nix [hunk 1]
```diff
@@ -18,7 +18,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = libintlOrEmpty ++
    [ intltool glib json_glib libsoup geoip
-     dbus dbus_glib avahi
+     dbus dbus_glib avahi wrapGAppsHook
```
Comment (jtojnar) [inline pkgs/development/libraries/geoclue/2.0.nix]: `wrapGAppsHook` should be moved to `nativeBuildInputs`. And while you are at it, can you also move `intltool` there?

**Review by jtojnar — COMMENTED** [review]
Builds fine, looks safe according to [changelog](https://cgit.freedesktop.org/geoclue/tree/NEWS?id=2.4.7).

```
$ maintainers/scripts/rebuild-amount.sh master
Estimating rebuild amount by counting changed Hydra jobs.
     14 x86_64-darwin
    104 x86_64-linux
```

File: pkgs/development/libraries/geoclue/2.0.nix (hunk 1 above)
Comment (ArdaXi) [reply pkgs/development/libraries/geoclue/2.0.nix]: Fixed, thanks!

**Review by jtojnar — APPROVED** [review]

**Comment by Mic92** [comment]
also backported to 17.09.

**Pull request merged**